src/overlay005/vs_seeker.c

```diff
@@ -0,0 +1,903 @@
+	{ 0x25B, 0x25B, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x25C, 0x25C, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x1, 0x2FD, 0x2FE, 0x2FF, 0x0, 0x0 },
+	{ 0x2, 0x2, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x4, 0x4, 0x0, 0x0, 0x0, 0x0 },
+	{ 0xA, 0xA, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x21, 0x21, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x163, 0x300, 0x301, 0xffff, 0x302, 0x0 },
+	{ 0x164, 0x164, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x6B, 0xffff, 0x303, 0xffff, 0x304, 0x0 },
+	{ 0x166, 0x166, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x167, 0x167, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x6C, 0xffff, 0xffff, 0x305, 0x307, 0x0 },
+	{ 0x168, 0x168, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x169, 0x169, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x44, 0xffff, 0xffff, 0x306, 0xffff, 0x308 },
+	{ 0x16A, 0xffff, 0xffff, 0xffff, 0xffff, 0x309 },
+	{ 0x25D, 0xffff, 0xffff, 0xffff, 0xffff, 0x30A }
+};
+
+static const MapObjectAnimCmd sVsSeekerAnimDoubleExclamationMark[] = {
+    {0x1, 0x1},
+    {0x67, 0x1},
+    {0xfe, 0x0}
+};
+
+static const MapObjectAnimCmd sVsSeekerAnimSingleExclamationMark[] = {
+    {0x4B, 0x1},
+    {0xfe, 0x0}
+};
+
+void VsSeeker_Start(TaskManager *taskMan, StringTemplate *template, u16 *outResult)
+{
+    FieldSystem *fieldSystem = TaskManager_FieldSystem(taskMan);
+    VsSeekerSystem *vsSeeker = Heap_AllocFromHeap(4, sizeof(VsSeekerSystem));
+
+    if (vsSeeker == NULL) {
+        GF_ASSERT(FALSE);
+        return;
+    }
+
+    memset(vsSeeker, 0, sizeof(VsSeekerSystem));
+
+    vsSeeker->fieldSystem = fieldSystem;
+    vsSeeker->events = SaveData_Events(fieldSystem->saveData);
+    vsSeeker->result = outResult;
+    vsSeeker->template = template;
+
+    sub_02050944(taskMan, VsSeeker_ExecuteTask, vsSeeker);
+    return;
+}
+
+static BOOL VsSeeker_ExecuteTask(TaskManager *taskMan)
+{
+    s32 missingBattery, numDigits;
+    enum VsSeekerUsability usability;
+    VsSeekerSystem *vsSeeker = TaskManager_Environment(taskMan);
+
+    switch (vsSeeker->state) {
+    case VS_SEEKER_STATE_WAIT_FOR_NPCS:
+        if (VsSeeker_WaitForNpcsToPause(vsSeeker->fieldSystem) == TRUE) {
+            VsSeekerSystem_SetState(vsSeeker, VS_SEEKER_STATE_CHECK_USABILITY);
+        }
+        break;
+    case VS_SEEKER_STATE_CHECK_USABILITY:
+        VsSeekerSystem_CollectViableNpcs(vsSeeker);
+        usability = VsSeekerSystem_CheckUsability(vsSeeker);
+
+        if (usability == VS_SEEKER_USABILITY_OK) {
+            *vsSeeker->result = VS_SEEKER_USE_RESULT_OK;
+            VsSeekerSystem_SetState(vsSeeker, VS_SEEKER_STATE_START);
+        } else if (usability == VS_SEEKER_USABILITY_NO_BATTERY) {
+            *vsSeeker->result = VS_SEEKER_USE_RESULT_NO_BATTERY;
+            VsSeekerSystem_SetState(vsSeeker, 7);
+        } else { // VS_SEEKER_USABILITY_NO_TRAINERS
+            *vsSeeker->result = VS_SEEKER_USE_RESULT_NO_TRAINERS;
+            VsSeekerSystem_SetState(vsSeeker, 8);
+        }
+        break;
+    case VS_SEEKER_STATE_START:
+        vsSeeker->playerStateTask = Player_SetStateVsSeeker(vsSeeker->fieldSystem);
+        Sound_PlayEffect(1568);
```

Comment:
Use a value from `constants/sdat.h` here (and, ideally, give it a good English name).